src/plot_api/plot_api.js

```diff
@@ -242,6 +242,24 @@ exports.plot = function(gd, data, layout, config) {
             fullLayout._glcanvas
                 .attr('width', fullLayout.width)
                 .attr('height', fullLayout.height);
+
+
+            var regl = fullLayout._glcanvas.data()[0].regl;
+            if(regl) {
+                if(
+                    fullLayout.width !== regl._gl.drawingBufferWidth ||
+                    fullLayout.height !== regl._gl.drawingBufferHeight
+                 ) {
+                    // Unfortunately, this can happen when relayouting to large
+                    // width/height on some browsers.
+                    Lib.log([
+                        'WebGL context buffer and canvas dimensions do not match,',
+                        'due to browser/WebGL bug.',
+                        'Clearing graph and plotting again.'
+                    ].join(' '));
+                    exports.newPlot(gd, gd.data, gd.layout);
```

Comment:
Hmm. This may not be the best idea actually. If the canvas and gl buffer dimensions still don't match after the replot, we could have an infinite loop.

Reply:
I'm not sure how that would happen, but if you're worried about it we could do something to limit to 1 redraw like
```
if(!gd._reglRedrawing) {
    gd._reglRedrawing = true;
    return exports.newPlot(...).then(function() { gd._reglRedrawing = false; });
}
```

Reply:
That's essentially when I went for in https://github.com/plotly/plotly.js/pull/2939/commits/85a646f374ab26b191a2c0eb9a1af470b0b72386